tests/users/test_dao_filter_by_skill.py

```diff
@@ -0,0 +1,130 @@
+        self.third_user = UserModel(
+            name=user3["name"],
+            email=user3["email"],
+            username=user3["username"],
+            password=user3["password"],
+            terms_and_conditions_checked=user3["terms_and_conditions_checked"],
+        )
+        self.third_user.is_email_verified = True
+        self.third_user.skills = "Critical thinking"
+
+        db.session.add(self.third_user)
+        db.session.commit()
+
+        # Insert data of the fourth entry
+        self.fourth_user = UserModel(
+            name=user4["name"],
+            email=user4["email"],
+            username=user4["username"],
+            password=user4["password"],
+            terms_and_conditions_checked=user4["terms_and_conditions_checked"],
+        )
+        self.fourth_user.is_email_verified = True
+        self.fourth_user.skills = "Creativity"
+
+        db.session.add(self.fourth_user)
+        db.session.commit()
+
+    def test_filter_users_by_skill_problem_solving(self):
+        self.insert_entries_in_database()
+
+        auth_header = get_test_request_header(self.admin_user.id)
+        expected_response = {"user1": "Problem Solving", "user2": "Problem Solving"}
+
+        actual_response = self.client.get(
+            "/users/verified?skills=problem solving",
+            headers=auth_header,
+            content_type="application/json",
+        )
+
+        self.assertEqual(HTTPStatus.OK, actual_response.status_code)
+
+        for data in json.loads(actual_response.data):
+
+            if data["username"] in expected_response.keys():
+                self.assertEqual(expected_response[data["username"]], data["skills"])
+
+    def test_filter_users_by_skill_critical(self):
+        self.insert_entries_in_database()
+
+        auth_header = get_test_request_header(self.admin_user.id)
+        expected_response = {"user3": "Critical thinking"}
+
+        actual_response = self.client.get(
+            "/users/verified?skills=Critical",
+            headers=auth_header,
+            content_type="application/json",
+        )
+
+        self.assertEqual(HTTPStatus.OK, actual_response.status_code)
+
+        for data in json.loads(actual_response.data):
+
+            if data["username"] in expected_response.keys():
+                self.assertEqual(expected_response[data["username"]], data["skills"])
+
+    def test_filter_users_by_skill_creative(self):
+        self.insert_entries_in_database()
+
+        auth_header = get_test_request_header(self.admin_user.id)
+        expected_response = {"user4": "Creativity"}
+
+        actual_response = self.client.get(
+            "/users/verified?skills=Creative",
+            headers=auth_header,
+            content_type="application/json",
+        )
+
+        self.assertEqual(HTTPStatus.OK, actual_response.status_code)
+
+        for data in json.loads(actual_response.data):
+
+            if data["username"] in expected_response.keys():
```

Comment:
This condition is not needed.

Reply:
I have added this change as per @devkapilbansal 's review comment. Same with other places.

Reply:
@devkapilbansal why is this required? If the data isn't present it should fail anyways. So we don't really need the condition right?

Reply:
Btw by this condition I meant line number 125.

Reply:
> I have added this change as per @devkapilbansal 's review comment. Same with other places.

Can you please link that review comment here. I am not able to find it

Reply:
If you are talking about [this one](https://github.com/anitab-org/mentorship-backend/pull/1097#discussion_r630489152), then I asked to assert if the user is correct.